### PR TITLE
[Paywalls] Remove lazy stack usages and fix alignment issues

### PR DESCRIPTION
**Pull request by MarkVillacampa**
Cramped three related things together in this PR:
- **Set the min length of `Spacer` to zero:** otherwise, there is a minimum size that depends on the platform but it's > 0 and could make a view overflow its parent if has fixed size and the sum of its child views + spacers exceeds it.
- **Removed usages of `LazyVStack` and `LazyHStack`:** they cannot have a child view with width/height type `fill` because the lazy rendering makes it impossible to know how much space they can occupy. There is not much potential for performance wins since the paywalls should be static and not have a lot of scrolling.
- **Fixed alignment issues to make rendering the view below possible:**

Screenshot 2024-11-22 at 17 52 41

3 alignments are needed to achieve this layout:
1- Align green box to the top of the red box
2- Align yellow boxes to the left of the green box.
3- Align yellow boxes to the top within each other.

I added code level comments to clarify what each alignment is doing, and PR comments on the lines where each of these three alignments are performed for extra clarity.

### CODE REVIEW

File: RevenueCatUI/Templates/V2/Components/Stack/FlexHStack.swift [hunk 1]
```diff
@@ -65,26 +65,23 @@ struct FlexHStack: View {
...
 
             case .spaceAround:
-                Spacer()
                 ForEach(0..<componentViewModels.count, id: \.self) { index in
+                    Spacer(minLength: 0)
                     ComponentsView(componentViewModels: [self.componentViewModels[index]], onDismiss: self.onDismiss)
-                    if index < self.componentViewModels.count - 1 {
-                        Spacer()
-                    }
+                    Spacer(minLength: 0)
```
Comment (MarkVillacampa) [inline RevenueCatUI/Templates/V2/Components/Stack/FlexHStack.swift:76]: Fixed the `spaceAround` flex layout: each items has a spacer around it, so there is two spacers between items and one at the start/end.

File: RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift [hunk 2]
```diff
@@ -125,17 +128,8 @@ struct VerticalStack: View {
...
-                alignment: horizontalAlignment.stackAlignment,
-                spacing: style.spacing
-            ) {
-                ComponentsView(
-                    componentViewModels: self.viewModels,
-                    onDismiss: self.onDismiss
-                )
-            }
+            // This alignment positions the items vertically within its parent
+            .frame(maxHeight: .infinity, alignment: distribution.verticalFrameAlignment)
```
Comment (MarkVillacampa) [inline RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift:132]: This is alignment number 1

File: RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift [hunk 3]
```diff
@@ -161,10 +155,16 @@ struct HorizontalStack: View {
 
     var body: some View {
         switch style.hstackStrategy {
-        case .normal, .lazy:
-            HStack(alignment: verticalAlignment.stackAlignment, spacing: style.spacing) {
+        case .normal:
+            HStack(
+                // This alignment positions inner items vertically relative to each other
```
Comment (MarkVillacampa) [inline RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift:160]: This is alignment number 3

File: RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift [hunk 4]
```diff
@@ -161,10 +155,16 @@ struct HorizontalStack: View {
...
+                alignment: verticalAlignment.stackAlignment,
+                spacing: style.spacing
+            ) {
                 ComponentsView(componentViewModels: self.viewModels, onDismiss: self.onDismiss)
             }
+            // This alignment positions the items horizontally within its parent
```
Comment (MarkVillacampa) [inline RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift:166]: This is alignment number 2: Align yellow boxes to the left of the green box.

File: RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift [hunk 5]
```diff
@@ -76,6 +77,7 @@ struct StackComponentView: View {
                     viewModels: self.viewModel.viewModels,
                     onDismiss: self.onDismiss
                 )
+                // This alignment positions the inner HStack vertically.
```
Comment (MarkVillacampa) [inline RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift:80]: This is alignment number 1: Align green box to the top of the red box

File: RevenueCatUI/Templates/V2/Components/Stack/StackComponentViewModel.swift [hunk 6]
```diff
@@ -136,39 +136,24 @@ struct StackComponentStyle {
...
 
-        // Not strategy for fit
-        switch self.size.width {
-        case .fit:
+        switch distribution {
+        case .start, .center, .end:
             return .normal
-        case .fill, .fixed:
+        case .spaceBetween, .spaceAround, .spaceEvenly:
             return .flex
```
Comment (MarkVillacampa) [inline RevenueCatUI/Templates/V2/Components/Stack/StackComponentViewModel.swift:157]: Flex stacks support `.start, .center, .end` distribution as well, so we could perfectly remove usages of VStack/HStack and leave just FlexVStack/FlexHStack.

However, the view structure of the latter are a bit more complex/nested than the simple stacks, so I've opted to keep them for now.

**Review by joshdholtz — APPROVED** [review]
This looks so good! :shipit: 

Side note... we should add the view like you have in the demo into a SwiftUI preview so we can have snapshots on it (at some point) 😁

**Pull request merged**